src/custom-elements/src/components/drawer/drawer.tsx

```diff
@@ -0,0 +1,243 @@
+import { Component, Event, EventEmitter, Method, Prop, State, Watch } from '@stencil/core';
+import { CARDINAL_POSITIONS } from '../../models/positions';
+import { ThreeStepsAnimation } from '../../utils/ThreeStepsAnimation';
+import { ANIMATION_DURATION, CLASSES } from '../../models/Constants';
+
+@Component({
+  tag: 'chi-drawer',
+  styleUrl: 'drawer.scss',
+  scoped: true
+})
+export class Drawer {
+
+  /**
+   * to set position of the drawer { top, right, bottom or left }
+   */
+  @Prop({ reflectToAttr: true }) position: string;
+
+  /**
+   * to add a backdrop behind the drawer, which covers the viewport
+   */
+  @Prop({ reflectToAttr: true }) backdrop: string;
+
+  /**
+   * to open or close the drawer
+   */
+  @Prop({ reflectToAttr: true, mutable: true }) active: boolean;
+
+  /**
+   * adds a close button
+   */
+  @Prop({ reflectToAttr: true }) collapsible: boolean;
+
+  /**
+   * to remove the space for the header
+   */
+  @Prop({ reflectToAttr: true }) headless: boolean;
+
+  /**
+   * header title. Not compatible with headless
+   */
+  @Prop({ reflectToAttr: true }) headerTitle: string;
+
+  /**
+   * Status classes for the show/hide animation
+   */
+  @State() _animationClasses: string;
+
+  /**
+   * Status classes for the show/hide animation
+   */
+  @State() _backdropAnimationClasses: string;
+
+  private animation: ThreeStepsAnimation;
+
+  @Watch('position')
+  positionValidation(newValue: string) {
+    if (newValue && !CARDINAL_POSITIONS.includes(newValue)) {
+      throw new Error(`Not valid position ${newValue} for drawer. Valid values are top, right, bottom or left. `);
+    }
+  }
+
+  @Watch('backdrop')
+  backdropValidation(newValue: string | boolean) {
+    if (newValue && newValue !== true && newValue !== 'inverse' && newValue !== 'backdrop') {
+      throw new Error(`Not valid backdrop ${newValue} for drawer. Valid values are inverse, backdrop, empty or true. `);
+    }
+  }
+
+  @Watch('active')
+  statusChanged(newValue: boolean, oldValue: boolean) {
+    // This weird double negation check is because of a stencil bug: https://github.com/ionic-team/stencil/issues/1238
+    if (!!newValue !== !!oldValue) {
+      if (newValue) {
+        this._show();
+      } else {
+        this._hide();
+      }
+    }
+  }
+
+  /**
+   * Toggles active state (show/hide)
+   */
+  @Method()
+  toggle() {
+    if (this.active) {
+      this.hide();
+    } else {
+      this.show();
+    }
+  }
+
+  /**
+   * Shows the drawer.
+   */
+  @Method()
+  show() {
+    this.active = true;
+  }
+
+  /**
+   * Hides the drawer
+   */
+  @Method()
+  hide() {
+    this.active = false;
+  }
```

Comment:
Ain't `show() and hide()` part of `toggle()` do we need to explicitly expose individual `show() and hide()` methods. @dani-cl-madrid any thoughts?

Reply:
@srinath-koneru Toggle is a shortcut for showing when hidden and hiding when shown. They individually make more sense than toggle itself, but I think they 3 are useful for non reactive development environments.
When using Angular, VueJS, or React, properties are more appropriate (we should make an integration page in the documentation), but when developing simple pages with no reactive templates, this methods are more useful, like the ones bootstrap uses in its components.